single_tech_samples/databricks/sample4_ci_cd/README.md

```diff
@@ -1 +1,256 @@
+
+1. Create the IaC Pipeline
+
+    In your Azure DevOps project, create a build pipeline from your forked repository:
+
+    ![build pipeline](images/build-connect.png "build pipeline")
+
+    Select the Existing Azure Pipelines YAML file option and set the path to [environment_setup/iac-create-environment-pipeline-arm.yml](environment_setup/iac-create-environment-pipeline-arm.yml)
+
+    > TODO add screenshot of the pipeline setup
+
+   Having done that, run the pipeline:
+
+   > TODO add screenshot of the pipeline run
+
+### 2.3. Deployed Resources
+
+  Check that the newly created resources appear in the [Azure Portal](https://portal.azure.com/):
+
+  The following resources will be deployed as a part of this template once the script is executed:
+
+  1. Azure Databricks workspace - development.
+
+  1. Azure Databricks workspace - staging.
+
+  1. Azure Databricks workspace - production.
+
+> TODO add screenshot
+
+### 2.4. Sample project Structure
+
+```txt
+
+├── devops
+|      ├──template
+|      |     └──jobs
+|      └──scripts
+├── multi-notebooks
+│   ├── README.md
+│   ├── devops
+│   │      ├──cd-pipeline.yml
+│   │      └──ci-pipeline.yml
+│   ├── notebooks
+│   │      ├── main_notebook.py
+│   │      ├── module_a_notebook.py
+│   │      └── module_b_notebook.py
+│   └── tests
+│          └── integration
+│               ├── main_notebook_test.py
+│               ├── module_a_test.py
+│               └── module_b_test.py
+├── notebook-dataframe
+│   ├── README.md
+│   ├── devops
+│   │      ├──cd-pipeline.yml
+│   │      └──ci-pipeline.yml
+│   ├── notebooks
+│   │      └──main_notebook.py
+│   └── tests
+│        └── integration
+│             └── main_notebook_test.py
+├── notebook-sparksql
+│   ├── README.md
+│   ├── devops
+│   │      ├──cd-pipeline.yml
+│   │      └──ci-pipeline.yml
+│   ├── notebooks
+│   │      └──main_notebook.py
+│   └── tests
+│        └── integration
+│             └── main_notebook_test.py
+├── notebook-python-lib
+│   ├── README.md
+│   ├── devops
+│   │      ├──cd-pipeline.yml
+│   │      └──ci-pipeline.yml
+│   ├── common
+│   │    ├── __init__.py
+│   │    ├── module_a.py
+│   │    └── module_b.py
+│   ├── notebooks
+│   │      └──main_notebook.py
+│   ├── setup.py
+│   └── tests
+│        ├── integration
+│        │    └── main_notebook_test.py
+│        └── unit
+│             ├── module_a_test.py
+│             └── module_b_test.py
+├── pyspark
+│   ├── README.md
+│   ├── devops
+│   │      ├──cd-pipeline.yml
+│   │      └──ci-pipeline.yml
+│   ├── common
+│   │    ├── __init__.py
+│   │    ├── module_a.py
+│   │    └── module_b.py
+│   ├── main.psy
+│   ├── setup.py
+│   └── tests
+│        ├── integration
+│        │     └── main_test.py
+│        └── unit
+│              ├── module_a_test.py
+│              └── module_b_test.py
+├── provision
```

Comment:
Since we are using devops pipelines and job templates for provisioning environment, should we add it to the "devops" folder itself? Or keep it as a separate "provision" folder?

Reply:
yes, I also change the folder diagram